numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1788,6 +1791,24 @@ unpack_bits(PyObject *input, int axis, PyObject *count_obj, char order)
         goto fail;
     }
 
+    /*
+     * setup lookup table under GIL, 256 8 byte blocks representing 8 bits
+     * expanded to 1/0 bytes
+     */
+    if (unpack_init == 0) {
+        npy_uint64 j;
+        for (j=0; j < 256; j++) {
+            npy_uint64 k;
+            for (k=0; k < 8; k++) {
+                npy_uint8 v = (j & (1 << k)) == (1 << k);
+                unpack_lookup_b[j][7 - k] = v;
+                /* for bitorder little the lookup table is byte swapped */
```

Comment:
I'd be tempted to attach this comment to bitorder big, since that's the one with the more complex expression